### PR TITLE
snap: refresh to core22

### PR DESCRIPTION
**Pull request by Saviq**
core24 needs to wait until Kodi 21 is available due to Python 3.12 incompatibility.

### CODE REVIEW

File: wayland-launch/bin/run-daemon [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+set -x
+
+if [ "$(snapctl get daemon)" = "true" ]
+then exec "$@"
+else
+  # It is unlikely, but possible for "snapctl stop ..." to fail
+  # temporarily (e.g. during install/refresh).
+  # Handle failure by ignoring it: it will work next retry.
+  # NB We don't move the logic to a `configure` hook, as that
+  # might conflict with any downstream snaps.
```
Comment (AlanGriffiths) [inline wayland-launch/bin/run-daemon]: There should be no downstream snap. ;)

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
Most of the text in the display seems to be missing

**Comment by Saviq** [comment]
> Most of the text in the display seems to be missing

That's https://github.com/canonical/mesa-core22/pull/20, added a workaround here.

**Review by AlanGriffiths — APPROVED** [review]
OK

**Pull request merged**